### ISSUE TITLE
Migrating from request to node-fetch

### ISSUE THREAD

**Issue by bureson** [restify/node-restify#1850]
Hello!

While this is not exactly restify related issue, we were wondering if could get some help for the issue that we are dealing with. Until early this year we were using `request` (https://github.com/request/request) to send requests to endpoints in our restify server. Since `request` is deprecated since 11th of February 2020, we migrated to `node-fetch` (https://github.com/node-fetch/node-fetch).

The deprecated package `request` had API for sending files inside the request (as formData) which behind the scenes allowed restify to serve the files under `req.files.file.path`, we are not able to achieve the same using `node-fetch`.

Syntax in `request`:

```
const formData = {
  file: {
    value: file,
    options: {
      filename: 'someFileName.xlsx',
      contentType: 'image/jpg'
    }
  }
}
const requestOptions = { ...., formData };
return requestPromiseFetch(url, requestOptions).then(response => { ....});
```
In `node-fetch` there is no specific syntax for sending files over HTTP, see https://github.com/node-fetch/node-fetch#post-with-form-data-detect-multipart and the formData is sent directly as body. It feels like the syntax in `node-fetch` is much closer to the actual request and extra work that `request` was doing behind the scenes has to be done manually, but whatever headers we set on the request, we were not able to make `restify` receive the file under `req.files`.

Could you please advise us what needs to be true in order for restify to put the file under req.files path?

**Comment by Talg123** [comment]
I think the main idea behind getting files on the server is to tell the server that the content-type is multipart/form-data
this way the server know it need to parse data in a different way then application/json etc...
I think that it has nothing to do with restify, you might need to add the content type on the request

**Comment by bureson** [comment]
@Talg123 I finally got around to investigate what exactly was wrong with my code. At some point I managed to post form-data as the request body:

```
const body = new FormData();
return fetch(url, { body, .... };
```

But that was not enough as the server recognized empty form data. To fix that I had to append an actual file to the new formData object. I'm surprised that this issue was kept open this whole time, but it's now solved :)
